src/modules/auth/auth.service.ts

```diff
@@ -257,4 +268,66 @@ export default class AuthenticationService {
       },
     };
   }
+
+  async requestSignInToken(requestSignInTokenDto: RequestSigninTokenDto) {
+    const { email } = requestSignInTokenDto;
+
+    const user = await this.userService.getUserRecord({ identifier: email, identifierType: 'email' });
+
+    if (!user) {
+      throw new NotFoundException({
+        message: USER_NOT_FOUND,
+        status_code: HttpStatus.NOT_FOUND,
+      });
+    }
+
+    const otpExist = await this.otpService.findOtp(user.id);
+
+    const newOtp = generateSixDigitToken();
+
+    if (otpExist) {
```

Comment:
If there is a situation that otpExist will be false, we should throw a descriptive error.

I checked the findOtp method, and it does not handle this as well.